### PR TITLE
refactor(coderd): move healthcheck report structs to codersdk

### PR DESCRIPTION
**Pull request by johnstcn**
Needed by https://github.com/coder/coder/issues/12161 to avoid an import cycle when adding a `codersdk.Client` method to hit `/api/v2/debug/health`.

Moves healthcheck report-related structs from `coderd/healthcheck` to `codersdk`

### CODE REVIEW

**Review by mtojek — COMMENTED** [review]
> +5,560 −5,548

Is it only because of the generator?

**Comment by johnstcn** [comment]
> > +5,560 −5,548
> 
> Is it only because of the generator?

Unfortunately, yes. Most of the changes are in generated files.

File: codersdk/health.go [hunk 1]
```diff
@@ -4,6 +4,13 @@ import (
 	"context"
 	"encoding/json"
 	"net/http"
+	"time"
+
+	"tailscale.com/derp"
+	"tailscale.com/net/netcheck"
+	"tailscale.com/tailcfg"
+
+	"github.com/coder/coder/v2/coderd/healthcheck/health"
```
Comment (mafredri) [inline codersdk/health.go:13]: I wonder if this also should be a part of codersdk. 🤔

**Review by mtojek — APPROVED** [review]
As long as linter does not complain about violating enterprise rules, it is 👍 from me.

**Comment by johnstcn** [comment]
Also smoke-tested that:
- Workspace builds OK
- Health page loads and renders OK in the UI

**Review by BrunoQuaresma — APPROVED** [review]

**Review by mafredri — APPROVED** [review]
Except for the removal of recover I flagged, LGTM!

File: coderd/healthcheck/database.go [hunk 2]
```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 	"time"
 
+	"github.com/coder/coder/v2/codersdk"
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply coderd/healthcheck/database.go:7]: yuck, will check and fix

File: coderd/healthcheck/derphealth/derp.go [hunk 3]
```diff
@@ -116,22 +71,16 @@ func (r *Report) Run(ctx context.Context, opts *ReportOptions) {
...
+			regionReport RegionReport
 		)
+		regionReport.Region = region
 		go func() {
 			defer wg.Done()
-			defer func() {
-				if err := recover(); err != nil {
-					regionReport.Error = ptr.Ref(fmt.Sprint(err))
-				}
-			}()
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply coderd/healthcheck/derphealth/derp.go:129]: good catch, part of a different refactor that I ended up reverting

File: scripts/apitypings/main.go [hunk 4]
```diff
@@ -28,13 +28,13 @@ import (
...
 	// baseDirs are the directories to introspect for types to generate.
-	baseDirs = [...]string{"./codersdk", "./coderd/healthcheck"}
+	baseDirs = [...]string{"./codersdk"}
 	// externalTypes are types that are not in the baseDirs, but we want to
 	// support. These are usually types that are used in the baseDirs.
 	// Do not include things like "Database", as that would break the idea
 	// of splitting db and api types.
 	// Only include dirs that are client facing packages.
-	externalTypeDirs = [...]string{"./cli/clibase", "./coderd/healthcheck/health", "./coderd/healthcheck/derphealth"}
+	externalTypeDirs = [...]string{"./cli/clibase", "./coderd/healthcheck/health"}
```
Comment (johnstcn) [inline scripts/apitypings/main.go:37]: review: no longer need to generate `coderd/healthcheck` and `coderd/healthcheck/derphealth`.

**Pull request merged**